packages/commonwealth/client/scripts/views/pages/members.tsx

```diff
@@ -1,121 +1,88 @@
+        return [...acc, ...page.results];
+      }, [] as SearchProfilesResponse['results'])
+      .map((p) => ({
+        id: p.id,
+        address_id: p.addresses?.[0]?.id,
+        address: p.addresses?.[0]?.address,
+        address_chain: p.addresses?.[0]?.chain,
+        chain: p.addresses?.[0]?.chain,
+        profile_name: p.profile_name,
+        avatar_url: p.avatar_url,
+        roles: p.roles,
+      }))
+      .map((p) => {
+        const minProfile = new MinimumProfile(p.address, p.chain);
+        minProfile.initialize(
+          p.profile_name,
+          p.address,
+          p.avatar_url,
+          p.id,
+          p.chain,
+          null
+        );
+        return {
+          profile: minProfile,
+          role: p.roles.find(
+            (role) =>
+              role.chain_id === app.activeChainId() &&
+              ['admin', 'moderator'].includes(role.permission)
+          ),
+        };
+      });
+  }, [data]);
 
-    if (response.totalCount) {
-      setTotalCount(response.totalCount);
-    }
+  const totalResults = data?.pages?.[0]?.totalResults || 0;
 
-    const members = response.profiles.map((p) => ({
-      id: p.id,
-      address_id: p.addresses?.[0]?.id,
-      address: p.addresses?.[0]?.address,
-      address_chain: p.addresses?.[0]?.chain,
-      chain_id: p.addresses?.[0]?.chain,
-      profile_name: p.profile_name,
-      avatar_url: p.avatar_url,
-      roles: p.roles,
-    }));
-
-    const profiles: Array<MemberInfo> = members.map((p) => {
-      const minProfile = new MinimumProfile(p.address, p.chain);
-      minProfile.initialize(
-        p.profile_name,
-        p.address,
-        p.avatar_url,
-        p.id,
-        p.chain,
-        null
-      );
-      return {
-        profile: minProfile,
-        role: p.roles.find(
-          (role) =>
-            role.chain_id === app.activeChainId() &&
-            ['admin', 'moderator'].includes(role.permission)
-        ),
-      };
-    });
-
-    if (reset) {
-      setMembersList(profiles);
-      if (containerRef) {
-        containerRef.current.scrollToIndex(0);
-      }
-    } else {
-      setMembersList([...membersList, ...profiles]);
-    }
-  };
-
-  // on debounced search term change, refresh search results
+  // fixes bug that prevents scrolling on initial page load
   useEffect(() => {
-    searchErr.current = null;
-    if (debouncedSearchTerm === '') {
-      handleLoadNextPage('', true);
+    const shouldFetchMore = members.length < 50 && totalResults > 50;
+    if (!shouldFetchMore) {
       return;
     }
-    if (debouncedSearchTerm.length >= 3) {
-      handleLoadNextPage(debouncedSearchTerm, true);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [debouncedSearchTerm]);
-
-  // on init, load first page
-  useEffect(() => {
-    handleLoadNextPage('', true);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    fetchNextPage();
+  }, [members, totalResults]);
```

Comment:
React Hook useEffect has a missing dependency: 'fetchNextPage'. Either include it or remove the dependency array